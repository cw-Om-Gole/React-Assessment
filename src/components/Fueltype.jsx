import React, { useState, useEffect } from 'react'
import "./FuelType.css";

const Fueltype = ({ fueltype, fuelNumber, setSelectedFuels, selectedFuels}) => {
    

    function capitalizeFirstLetter(word) {
        if (!word) return '';
        return word.charAt(0).toUpperCase() + word.slice(1);
    }

    const handleCheck = (e) => {
      const { checked } = e.target;
      setSelectedFuels(prevState => {
        if(checked) {
          return [...prevState, fuelNumber];
        } else {
          return prevState.filter(value => value != fuelNumber);
        }
      })
    }



  return (
    <div className="checkbox-container">
        <input type="checkbox" id={fueltype} name={fueltype} onChange={handleCheck} checked={selectedFuels.includes(fuelNumber)}/>
        <label for={fueltype}>{capitalizeFirstLetter(fueltype)}</label>
    </div>
  )
}

export default Fueltype;
