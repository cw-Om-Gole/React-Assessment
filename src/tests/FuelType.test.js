import React from 'react';
import { render, screen ,fireEvent  } from '@testing-library/react';
import Fueltype from '../components/Fueltype';

describe('Fueltype Component Snapshot', () => {
    const mockSetSelectedFuels = jest.fn();

    const mockProps = {
        fueltype: 'diesel',
        fuelNumber: 1,
        setSelectedFuels: mockSetSelectedFuels,
        selectedFuels: [1],
    };

    it('matches the snapshot when checkbox is checked', () => {
        const { asFragment } = render(
            <Fueltype 
                fueltype={mockProps.fueltype} 
                fuelNumber={mockProps.fuelNumber} 
                setSelectedFuels={mockProps.setSelectedFuels} 
                selectedFuels={mockProps.selectedFuels} 
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it('matches the snapshot when checkbox is unchecked', () => {
        const { asFragment } = render(
            <Fueltype 
                fueltype={mockProps.fueltype} 
                fuelNumber={mockProps.fuelNumber} 
                setSelectedFuels={mockProps.setSelectedFuels} 
                selectedFuels={[]} 
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });

    test('should render checkbox with label', () => {
        render(
            <Fueltype
                fueltype="petrol"
                fuelNumber="1"
                setSelectedFuels={mockProps.setSelectedFuels}
                selectedFuels={[]}
            />
        );

        const checkbox = screen.getByLabelText('Petrol');
        expect(checkbox).toBeInTheDocument();
        expect(checkbox).toHaveAttribute('type', 'checkbox');
    });

    test('Check if setSelectedFuels is called', () => {
        render(
            <Fueltype
                fueltype={mockProps.fueltype} 
                fuelNumber={mockProps.fuelNumber} 
                setSelectedFuels={mockProps.setSelectedFuels}
                selectedFuels={[]}
            />
        );

        const checkbox = screen.getByRole('checkbox');
        fireEvent.click(checkbox); 
        fireEvent.click(checkbox); 

        expect(mockProps.setSelectedFuels).toHaveBeenCalledTimes(2);
    });
});
