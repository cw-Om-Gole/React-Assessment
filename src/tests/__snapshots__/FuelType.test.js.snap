// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fueltype Component Snapshot matches the snapshot when checkbox is checked 1`] = `
<DocumentFragment>
  <div
    class="checkbox-container"
  >
    <input
      checked=""
      id="diesel"
      name="diesel"
      type="checkbox"
    />
    <label
      for="diesel"
    >
      Diesel
    </label>
  </div>
</DocumentFragment>
`;

exports[`Fueltype Component Snapshot matches the snapshot when checkbox is unchecked 1`] = `
<DocumentFragment>
  <div
    class="checkbox-container"
  >
    <input
      id="diesel"
      name="diesel"
      type="checkbox"
    />
    <label
      for="diesel"
    >
      Diesel
    </label>
  </div>
</DocumentFragment>
`;
